src/main/java/fi/helsinki/opintoni/service/AvatarImageService.java

```diff
@@ -46,14 +47,15 @@ public String getAvatarImageUrl(Long userId) {
     }
 
     private String getAvatarImageUrl(Supplier<String> defaultAvatarUrlSupplier, Long userId) {
-        UserSettings userSettings = userSettingsRepository.findByUserId(userId);
+        UserSettings userSettings = userSettingsRepository.findByUserId(userId).orElseThrow(NotFoundException::new);
```

Comment:
Olisiko hyvä, jos repossa olisi kaksi metodia, joista toinen heittää exceptin jos ei löydä, ja toinen ei. Niin ei tarvi näin monessa paikassa käsitellä löytymättömyyttä.
Esim findByUserId vs. searchByUserId.